### DISCUSSION TITLE
[Feature] Ability to set custom status via DurableOrchestrationClient

### DISCUSSION THREAD

**Discussion by brandonh-msft** [Azure/azure-functions-durable-extension#1862] [Ideas]
In the following code:
```csharp
log.Info(@"Starting orchestrator...");
var newInstanceId = await starter.StartNewAsync(nameof(MyOrchestrationFunction), null);
log.Info($@"- Instance id: {newInstanceId}");

return starter.CreateCheckStatusResponse(req, newInstanceId);
```
I'd like to be able to set the custom status so calls to the GetStatus API can return better information while waiting for the orchestrator to pick up and begin execution.

Today I can only set custom status off `DurableOrchestrationContext` which means I have to wait for the orchestrator to start running before I can return anything meaningful.

**Comment by cgillum** [comment]
Can you go into more detail about why you might need custom status in your scenario? My initial reaction is that a runtime status value of "Pending" should be sufficient to communicate that the orchestration hasn't started yet.

**Comment by brandonh-msft** [comment]
I guess in my mind Pending didn't tell me that the orchestration had been successfully queued for execution. Also I was keying off the CustomStatus value and started hitting instances where it came back `null` despite the POST request to kick off the Function succeeding and giving me back the status URLs. I'd like the opportunity to populate it, I guess.

**Comment by cgillum** [comment]
It might be possible to do this if we make a change to the durable task framework. I'd like to hear if others would like a similar feature though.
